manim/animation/composition.py

```diff
@@ -59,6 +59,8 @@ def finish(self) -> None:
 
     def clean_up_from_scene(self, scene: Scene) -> None:
         for anim in self.animations:
+            if self.remover:
+                anim.remover = True
```

Comment:
> IMO this really isn't the right place to set this attribute especially because `Scene.compile_animations` handles setting the `play` kwargs as the attributes of its animations.

Right, passing the `play` kwargs, i.e. `remover`, to an animation which happened to be an AnimationGroup didn't do anything since the AnimationGroup never did anything with that kwarg.

> Also, `AnimationGroup`s can be nested inside other `AnimationGroup`s, so we'll have to use the recursive approach. Take this code for example:
> ```python
> class RemoverTest(Scene):
>     def construct(self):
>         s = Square()
>         c = Circle()
>         self.play(
>             AnimationGroup(
>                 AnimationGroup(Create(s), Create(RegularPolygon(5).to_edge(UP))),
>                 Write(c, remover=True),
>             ),
>             remover=True,
>         )
>         self.wait(0.1)
> ```

The output on my side is that those mobjects are gone as expected with my changes?

https://user-images.githubusercontent.com/44864613/123387135-bc3fae00-d597-11eb-94a0-82db788a6a4a.mp4

> The above code block in `Scene.compile_animations` can be replaced with something like so:
> 
> ```python
> from manim.animation.composition import AnimationGroup  # import here to avoid circular imports
> 
> def update_anim_props(animation):
>     if isinstance(animation, AnimationGroup):
>         for anim in animation.animations:
> 	    update_anim_props(anim)
>     for k, v in kwargs.items():
> 	setattr(animation, k, v)
> 
> for animation in animations:
>     update_anim_props(animation)
> ```

I like your approach better than mine. Let me think about it a little further to fully understand it. Cheers for the pointers!

Reply:
> The output on my side is that those mobjects are gone as expected with my changes?

Ahh I'm sorry I thought I was testing on your branch but was actually on a different (similar) personal branch. You are correct, the mobjects are all gone with your changes. Although I do think the relevant code should be migrated to `Scene.compile_animations`.

Reply:
> Although I do think the relevant code should be migrated to `Scene.compile_animations`.

I understand. My initial thought process was that passing kwargs to an AnimationGroup should probably be handled by a method in AnmationGroup itself. But I guess handling the kwargs together with all the kwargs of `play` in `Scene.compile_animations` makes more sense.

Reply:
I just found out that your suggestion as is wouldn't get this to work properly:
```py
class RemoverTest2(Scene):
    def construct(self):
        s = Square()
        c = Circle()
        self.play(
            AnimationGroup(
                Create(s),
                Write(c),
                remover=True
            )
        )
```
since the changes in `Scene.compile_animations` would add any kwargs of `play` to the animations in the group but they don't  fix the issue that kwargs set in AnimationGroup don't get added to the single animations.

Reply:
> they don't fix the issue that kwargs set in AnimationGroup don't get added to the single animations.

Ah, I missed that, thanks for pointing out! I replaced the `update_anim_props` function with the following and it works. I think this should work well with all sorts of combinations of animations now. Let me know if you find any other issues :)
```py
def update_anim_props(animation):
    if isinstance(animation, AnimationGroup):
        for anim in animation.animations:
            anim.remover = animation.is_remover()
	    update_anim_props(anim)
    for k, v in kwargs.items():
	setattr(animation, k, v)
```

Reply:
Cheers! Just for my peace of mind if you don't mind: This passes everything directly set in `play` directly to the animations in the AnimationGroups. Do we never want that something in `play` gets passed only to the AnimationGroup instead of passing it right through?

Reply:
> Do we never want that something in `play` gets passed only to the AnimationGroup instead of passing it right through?

Currently `lag_ratio` is one such parameter that doesn't make its way into individual animations in an `AnimationGroup`. In this example:
```py
class Test(Scene):
    def construct(self):
        vg = VGroup(Square(), Circle())
        rp = RegularPolygon(5).to_edge(UP)
        self.play(
            AnimationGroup(
                Create(vg, lag_ratio=0.5),
                Create(rp),
                lag_ratio=1,
                remover=False
            ),
            lag_ratio=0,  # doesn't apply to AnimationGroup as a whole or to its individual anims 
            remover=True  # applies to individual anims in AnimationGroup
        )
        self.wait()
```
`remover=True` is applied to every individual animation in `AnimationGroup`, whereas the `lag_ratio=0` affects neither the individual animations of `AnimationGroup`, nor the group as a whole. 

https://user-images.githubusercontent.com/73361366/123501555-6b958700-d663-11eb-8158-ee7faa7f1f49.mp4

But with my suggestion the `lag_ratio=0` affects every individual animation (all animations run in parallel), which I think isn't good.

So the conclusion I've come to is that the PR is good to go in its current state, thanks for sharing your thoughts! I think the discussion once again points at the `Animation` system in manim being really whacky and needs a rewrite in the future. The fact that only some play kwargs affect individual animations and others don't isn't good at all IMO.